contracts/DxMgnPool.sol

```diff
@@ -30,7 +46,7 @@ contract DxMgnPool {
     function deposit(uint amount) public {
         uint poolShares = calculatePoolShares(amount);
         Participation memory participation = Participation({
-            startAuctionId: dx.getAuctionIndex(address(depositToken), address(secondaryToken)),
```

Comment:
I think, we should always get the auction index from the exchange.
This would have the benefit that if we miss one auction,(bots dead) the system will not break.

(not sure whether you did it just for testing...)

Reply:
This exact thought let me to change this. Imagine our bot forgot to participate in an auction. This means there will be no MGN generated. However, this logic would make it seem as if we did generate MGN (e.g. a newly joined depositor will get some credit, although his funds were not yet used).

Therefore having an internal counter, that we increment whenever we actually claim MGN and participate in a new auction seemed safer to me.

Reply:
Nice, yeah. 

After progressing with my code I saw the same thing.